react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -54,23 +44,30 @@ class App extends Component {
 
   render() {
     return (
-      <Fragment>
-        <Navbar />
-        <div className={styles.container}>
-          <Search onSearch={this.onSearch} />
-          {this.state.books.length ? (
-            this.state.books.map(this.renderBooks)
-          ) : (
-            <div className={styles.noData}>
-              <h2 className={styles.title}>No Data</h2>
-            </div>
-          )}
-        </div>
-        {this.state.bookSelected.length ? (
-          <ShoppingCart data={this.state.bookSelected} addItem={this.addItem} removeItem={this.removeItem} />
-        ) : null}
-        <Footer />
-      </Fragment>
+      <Provider store={store}>
+        <Fragment>
+          <Navbar />
+          <Book key={this.state.books[0].id} data={this.state.books[0]} configButton={undefined} />
+          <div className={styles.container}>
+            <Search onSearch={this.onSearch} />
+            {this.state.books.length ? (
+              this.state.books.map(this.renderBooks)
+            ) : (
+              <div className={styles.noData}>
+                <h2 className={styles.title}>No Data</h2>
+              </div>
+            )}
+          </div>
+          {this.state.bookSelected.length ? (
```

Comment:
`this.state.bookSelected.length !== 0 && (...)`

If this.state.bookSelected.length === 0, the next part of the and isn't executed and it returns `false`. `false`, `null` and `undefined` aren't rendered, so it's fine to do it this way